internal/cloud/state.go

```diff
@@ -400,8 +400,17 @@ func (s *State) Unlock(id string) error {
 }
 
 // Delete the remote state.
-func (s *State) Delete() error {
-	err := s.tfeClient.Workspaces.Delete(context.Background(), s.organization, s.workspace.Name)
+func (s *State) Delete(force bool) error {
+
+	var err error
+
+	isSafeDeleteSupported := s.workspace.Permissions.CanForceDelete != nil
```

Comment:
You can temporarily update go.mod with the branch version of go-tfe to test this while it's a PR. Just be sure to get go-tfe merged and released first and reset the go.mod dependency to the new version before merging this PR.